acvm/src/pwg/directives.rs

```diff
@@ -112,3 +103,48 @@ pub fn solve_directives(
         }
     }
 }
+
+fn to_radix_outcome(
+    witnesses: &Vec<Witness>,
+    decomposed_integer: Vec<u8>,
+    initial_witness: &mut BTreeMap<Witness, FieldElement>,
+    is_little_endian: &bool,
+) -> Result<(), OpcodeResolutionError> {
+    if witnesses.len() < decomposed_integer.len() {
+        return Err(OpcodeResolutionError::UnsatisfiedConstrain);
+    }
+
+    if *is_little_endian {
+        for (i, witness) in witnesses.into_iter().enumerate() {
+            // Fetch the `i'th` digit from the decomposed integer list
+            // and convert it to a field element.
+            // If it is not available, which can happen when the decomposed integer
+            // list is shorter than the witness list, we return 0.
+            let value = match decomposed_integer.get(i) {
+                Some(digit) => FieldElement::from_be_bytes_reduce(&[*digit]),
+                None => FieldElement::zero(),
+            };
+
+            insert_value(witness, value, initial_witness)?
+        }
+    } else {
+        // if it is big endian and the decompoased integer list is shorter
+        // than the witness list, pad the extra part with 0 first then
+        // add the decompsed interger list to the witness list.
+        let padding_len = witnesses.len() - decomposed_integer.len();
+        for (i, witness) in witnesses.into_iter().enumerate() {
+            if i < padding_len {
```

Comment:
You can return a `FieldElement` from this if statement, then call `insert_value` once after rather than in both if and else cases.